### DISCUSSION TITLE
Using Darkmode with Keycloakify

### DISCUSSION THREAD

**Discussion by chmcquee** [InseeFrLab/onyxia#494] [Q&A] [answered]
Hi again,
I'm searching for a custom dark mode implementation that works in production with keycloakify and URL params. I don't need to switch the dark mode just get the values.
I see there's a dark mode switch on the homepage of this app, but I can't find the file where it's implemented.
Can you help me out?
Thanks in advance 👍 
Chris

**Comment by garronej** [comment, marked as answer]
Unless you planning to use https://github.com/InseeFrLab/onyxia-ui it's not rellevent to try to reverse engeneer how the dark mode works.  

You'll be beter off reading this: https://mui.com/material-ui/customization/dark-mode/  

https://github.com/garronej/powerhooks/blob/main/src/test/spa/src/TestUseGlobalState/useIsDarkModeEnabled.tsx can be used to switch the global state and persist across reloads.

**Reply by chmcquee** [reply]
True makes sense. Thanks for your comment and explanation @garronej
